catalog/resource_metastore.go

```diff
@@ -65,6 +65,7 @@ func ResourceMetastore() *schema.Resource {
 			var update catalog.UpdateMetastore
 			common.DataToStructPointer(d, s, &create)
 			common.DataToStructPointer(d, s, &update)
+			update.Name = d.Id()
```

Comment:
id won't be set yet in Create

```suggestion
			update.Name = d.Get("name")
```

Reply:
Good catch